huaweicloud/services/cci/resource_huaweicloud_cciv2_deployment.go

```diff
@@ -0,0 +1,1433 @@
+	createV2DeploymentOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"Content-Type": "application/json"},
+	}
+	createV2DeploymentOpt.JSONBody = utils.RemoveNil(buildCreateV2DeploymentParams(d))
+
+	resp, err := client.Request("POST", createV2DeploymentPath, &createV2DeploymentOpt)
+	if err != nil {
+		return diag.Errorf("error creating CCI V2 Deployment: %s", err)
+	}
+
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	ns := utils.PathSearch("metadata.namespace", respBody, "").(string)
+	name := utils.PathSearch("metadata.name", respBody, "").(string)
+	if ns == "" || name == "" {
+		return diag.Errorf("unable to find CCI V2 Deployment name or namespace from API response")
+	}
+	d.SetId(ns + "/" + name)
+
+	err = waitForV2DeploymentStatusToCompleted(ctx, client, ns, name, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return resourceV2DeploymentRead(ctx, d, meta)
+}
+
+func buildCreateV2DeploymentParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"kind":       "Deployment",
+		"apiVersion": "cci/v2",
+		"metadata": map[string]interface{}{
+			"name":      d.Get("name"),
+			"namespace": d.Get("namespace"),
+		},
+		"spec": map[string]interface{}{
+			"replicas":                d.Get("replicas"),
+			"minReadySeconds":         utils.ValueIgnoreEmpty(d.Get("min_ready_seconds")),
+			"progressDeadlineSeconds": utils.ValueIgnoreEmpty(d.Get("progress_deadline_seconds")),
+			"selector":                buildLabelSelectorParams(d.Get("selector.0")),
+			"template":                buildV2DeploymentTemplateParams(d.Get("template.0")),
+			"strategy":                buildV2DeploymentStrategyParams(d.Get("strategy.0")),
+		},
+	}
+
+	return bodyParams
+}
+
+func buildV2DeploymentStrategyParams(strategy interface{}) map[string]interface{} {
+	if strategy == nil {
+		return nil
+	}
+
+	return map[string]interface{}{
+		"type":           utils.PathSearch("type", strategy, nil),
+		"rolling_update": utils.PathSearch("rolling_update", strategy, nil),
+	}
+}
+
+func buildV2DeploymentTemplateParams(template interface{}) map[string]interface{} {
+	if template == nil {
+		return nil
+	}
+
+	metadata := utils.PathSearch("metadata|[0]", template, nil)
+	spec := utils.PathSearch("spec|[0]", template, nil)
+	return map[string]interface{}{
+		"metadata": buildV2DeploymentTemplateMetadataParams(metadata),
+		"spec":     buildV2DeploymentTemplateSpecParams(spec),
+	}
+}
+
+func buildV2DeploymentTemplateSpecParams(spec interface{}) map[string]interface{} {
+	if spec == nil {
+		return nil
+	}
+
+	containers := utils.PathSearch("containers", spec, make([]interface{}, 0)).([]interface{})
+	affinity := utils.PathSearch("affinity|[0]", spec, nil)
+	imagePullSecrets := utils.PathSearch("image_pull_secrets", spec, nil).(*schema.Set).List()
```

Comment:
```suggestion
	imagePullSecrets := utils.PathSearch("image_pull_secrets", spec, &schema.Set{}).(*schema.Set).List()
```
and fix other places